contracts/wallet/BaseWallet.sol

```diff
@@ -101,16 +101,25 @@ contract BaseWallet {
      * @param _value The value of the transaction.
      * @param _data The data of the transaction.
      */
-    function invoke(address _target, uint _value, bytes calldata _data) external moduleOnly {
+    function invoke(address _target, uint _value, bytes calldata _data) external moduleOnly returns (bytes memory _result) {
+        bool success;
         // solium-disable-next-line security/no-call-value
-        (bool success, ) = _target.call.value(_value)(_data);
-        require(success, "BW: call to target failed");
+        (success, _result) = _target.call.value(_value)(_data);
+        if(!success) {
+            // solium-disable-next-line security/no-inline-assembly
+            assembly {
+                // returndata = {sig:4}{pos:32}{revert_reason_length:32}{revert_reason} as per EIP838
+                let len := sub(returndatasize, 0x44)
+                returndatacopy(0, 0x44, len)
+                revert(0, len)
+            }
```

Comment:
I still don't understand why we don't do:
```
assembly {
                returndatacopy(0, 0, returndatasize)
                revert(0, returndatasize)
            }
``` 
and return the entire byte array from the error we are bubbling up?

Reply:
Sorry, I didn't see your previous response at it was outdated and hidden.

Still, isn't it safer to return exactly what the lower call error is returning and let the client deal with any potential junk?

Reply:
You're right. I got confused by an incorrect idea that every subsequent revert would recursively add some more junk... (that's not the case using `assembly{revert()}`). Changed.